src/coldcard.js

```diff
@@ -507,107 +516,177 @@ export class ColdcardSignMultisigTransaction extends ColdcardInteraction {
+   * Derivation: m/45'
+   * Format: P2WSH
+   * 
+   * D0CFA66B: tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9
+   * 8E697B74: tpubD97nVL37v5tWyMf9ofh5rzn...XgSc
+   * BE26B07B: tpubD9ArfXowvGHnuECKdGXVKDM...FxPa
+   * 4369050F: tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n
+   * 
+   * @param {string} configString valid coldcard config string
+   * @returns {MultisigWalletConfig} new instance of MultisigWalletConfig
+   * @example
+   * import { parseColdcardConfig } from "unchained-wallets";
+   * 
+   * const config = MultisigWalletConfig.fromColdCardConfig(ccFileData);
+   * console.log(JSON.parse(config.toJSONConfig()))
+   * 
+   * {
+   *    name: MeMyself,
+   *    quorum: {
+   *      totalSigners: 4,
+   *      requiredSigners: 2,   
+   *    },
+   *    addressType: 'P2WSH',
+   *    network: 'testnet',
+   *    extendedPublicKeys: [
+   *      {
+   *        xfp: 'D0CFA66B',
+   *        xpub: 'tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9'
+   *      },
+   *      {
+   *        xfp: '8E697B74',
+   *        xpub: 'tpubD97nVL37v5tWyMf9ofh5rzn...XgSc',
+   *      },
+   *      {
+   *        xfp: 'BE26B07B',
+   *        xpub: 'tpubD9ArfXowvGHnuECKdGXVKDM...FxPa',
+   *      },
+   *      {
+   *        xfp: '4369050F',
+   *        xpub: 'tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n',
+   *      }
+   *    ]
+   * }
+   */
+export function parseColdcardConfig(configString) {
+    if (typeof configString !== 'string') {
+      throw new TypeError(
+        `ColdCard config must be a string, instead received ${typeof configString}`
+      );
     }
-
-    if (this.jsonConfig.extendedPublicKeys &&
-      this.jsonConfig.extendedPublicKeys.every((xpub) => {
-        // For each xpub, check that xfp exists, the length is 8, type is string, and valid hex
-        if (!xpub.xfp || xpub.xfp === "Unknown") {
-          throw new Error("ExtendedPublicKeys missing at least one xfp.");
+    
+    const options = {};
+    // split the text by line
+    const lines = 
+      configString
+        .split('\n')
+        // formatting
+        .map(line => line.trim())
+        // ignore comments
+        .filter(line => line[0] !== '#' && line.length);
+
+    // for each line separate key and values by colon
+    lines.forEach(line => {
+      // trim white space and set to lowercase
+      const [key, value] = line.split(':').map(str => str.trim());
+      // handle recognized keys (name, format, derivation, and policy)
+      switch (key.toLowerCase()) {
+        case 'name':
+          options.name = value;
+          break;
+        case 'format':
+          options.addressType = value.toUpperCase();
+          break;
+        case 'derivation': {
+          const pathError = validateBIP32Path(value);
+          assert(!pathError.length, pathError);
+          options.derivation = value;
+          break;
         }
-        if (typeof xpub.xfp !== 'string') {
-          throw new Error("XFP not a string");
+        case 'policy': {
+          let [m, n] = value.split(' of ');
```

Comment:
this can be a problem because coldcard supports many varieties of representing this

see https://github.com/Coldcard/firmware/blob/master/shared/multisig.py#L580

Reply:
To be clear, you're referencing https://github.com/Coldcard/firmware/blob/master/shared/multisig.py#L637 I'm guessing? It looks like this should be a regex instead then, just splitting up two digits.